File: perfectnumber.cpp
#include <iostream>
using namespace std;
int main()
{
    int n,i=1,sum=0;
    cout<<"Enter a number to check if its a perfect number: ";
    cin>>n;
    while(i<n)
    {
        if(n%i==0)
        {
            sum=sum+i;
        }
        i++;
    }
    if(sum==n)
    {
        cout<<i<<" is a perfect number";
    }
    else{
        cout<<i<<" is not a perfect number";
    }
}

File: perfectnumber.py
n=int(input("Enter a number to check if it is perfect number or not: "))
sum=0
for x in range(1,n):
    if n%x==0:
        sum=sum+x
if(sum==n):
    print(n," is a perfect number")
else:
    print(n," is not a perfect number")